nifi-nar-bundles/nifi-media-bundle/nifi-media-processors/src/main/java/org/apache/nifi/processors/document/ExtractDocumentText.java

```diff
@@ -0,0 +1,142 @@
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.tika.Tika;
+import org.apache.tika.exception.TikaException;
+
+import java.io.BufferedInputStream;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+
+@Tags({"extract, text, pdf, word, excel, powerpoint, office"})
+@CapabilityDescription("Run Apache Tika text extraction to extra the text from supported binary file formats such as PDF " +
+        "and Microsoft Office files.")
+public class ExtractDocumentText extends AbstractProcessor {
+    private static final String TEXT_PLAIN = "text/plain";
+
+    public static final String FIELD_MAX_TEXT_LENGTH = "MAX_TEXT_LENGTH";
+    public static final String FIELD_SUCCESS = "success";
+    public static final String FIELD_FAILURE = "failure";
+
+    public static final PropertyDescriptor MAX_TEXT_LENGTH = new PropertyDescriptor.Builder()
+            .name(FIELD_MAX_TEXT_LENGTH)
+            .displayName("Max Output Text Length")
+            .description("The maximum length of text to retrieve. This is used to limit memory usage for " +
+                    "dealing with large files. Specify -1 for unlimited length.")
+            .required(false).defaultValue("-1").addValidator(StandardValidators.INTEGER_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES).build();
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder().name(FIELD_SUCCESS)
+            .description("Successfully extract content.").build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder().name(FIELD_FAILURE)
+            .description("Failed to extract content.").build();
+
+    private List<PropertyDescriptor> descriptors = Collections.unmodifiableList(Arrays.asList(MAX_TEXT_LENGTH));
+    private Set<Relationship> relationships = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(REL_SUCCESS, REL_FAILURE)));
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        return;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            flowFile = session.create();
```

Comment:
This should be changed to a short-circuit return instead of creating a new empty FlowFile.
```suggestion
            return;
```